R/select.R

```diff
@@ -0,0 +1,333 @@
+    return(table.out)
+}
+
+selectFromColumns <- function(table, table.orig, selection.mode = "vector",
+                           selections = NULL, unit, calendar, ...)
+{
+    n.dims <- length(dim(table))
+    if (selection.mode == "range"){
+        selections <- parseRangeString(selections)
+        selections <- checkNumericSelections(table, selections, 2)
+    }else if (selection.mode == "date range")
+        selections <- findDatesInTable(table, selections, 2)
+
+    if (selection.mode == "first columns"){
+        table.out <- First(table, keep = selections, unit = "Column", ...)
+        selections <- seq_len(ncol(table.out))
+    }else if (selection.mode == "last columns"){
+        table.out <- Last(table, keep = selections, unit = "Column", ...)
+        selections <- (ncol(table) - ncol(table.out) + 1):ncol(table)
+    }else if (selection.mode == "first date-time periods"){
+        table.out <- t(First(t(table),  # t() in case date labels in rows and columns
+                             keep = selections, unit = unit,
+                             calendar = calendar, ...))
+        selections <- seq_len(ncol(table.out))
+    }else if (selection.mode == "last date-time periods"){
+        table.out <- t(Last(t(table),  # t() in case date labels in rows and columns
+                             keep = selections, unit = unit,
+                            calendar = calendar, ...))
+        selections <- (ncol(table) - ncol(table.out) + 1):ncol(table)
+    }else
+    {
+        if (is.character(selections))
+            selections <- checkCharacterSelections(table, selections, 2)
+        else
+            selections <- checkNumericSelections(table, selections, 2)
+        if (length(dim(table)) == 2)
+            table.out <- table[, selections, drop = FALSE]
+        else if (length(dim(table)) == 3)
+            table.out <- table[, selections, , drop = FALSE]
+    }
+    if (NCOL(table.out) == 0L)
+        stop("No columns selected, output contains no columns.")
+
+    table.out <- flipU::CopyAttributes(table.out, table)
+    if (hasColSpan(table))
+         table.out <- updateTableColSpanAttribute(table.out, table, selections)
+    return(table.out)
+}
+
+checkCharacterSelections <- function(table, indices, dim)
+{
+    indices <- unique(indices)
+    if (is.data.frame(table))
+    {
+        if (dim == 1)
+            names <- names(table)
+        else
+            names <- rownames(table)
+    }else
+        names <- dimnames(table)[[dim]]
+
+    bad.idx <- which(!indices %in% names)
+    if (length(bad.idx))
+    {
+        dim.str <- ifelse(dim == 1, "row", "column")
+        if (length(bad.idx) == length(indices))
+            stop("No valid selections were found in the ", dim.str,
+                 " labels. If supplying selections using a page control, check ",
+                 " the item list for the control and edit if necessary.",
+                 call. = FALSE)
```

Comment:
This doesn't seem to make a difference? I still see the call details when I tested this

Reply:
Hmm, not sure what's happening there. It's been a documented feature of base R for a long time. Did you try outside RStudio or in Displayr?

Reply:
In Displayr, maybe Displayr ignores the parameter